toucan_connectors/workday/workday_connector.py

```diff
@@ -0,0 +1,153 @@
+    )
+    request_criteria_param: dict = Field(
+        None,
+        title='Request Criteria Parameter',
+        description='Check https://community.workday.com/sites/default/files/file-hosting/productionapi/index.html',
+        example="{'Exclude_Inactive_Workers' : False}"
+    )
+    response_filter_param: dict = Field(
+        {},
+        title='Response Filter Parameter',
+        description='Check https://community.workday.com/sites/default/files/file-hosting/productionapi/index.html',
+        example="{'As_Of_Effective_Date': effective_date,'As_Of_Entry_DateTime': effective_date}"
+    )
+    response_group_param: dict = Field(
+        None,
+        title='Response Group Parameter',
+        description='Check https://community.workday.com/sites/default/files/file-hosting/productionapi/index.html',
+        example="{'Include_Reference' : True,'Include_Absence_Input_Data' : True}"
+    )   
+    filter: str = FilterSchema
+
+
+class WorkdayConnector(ToucanConnector):
+    data_source_model: WorkdayDataSource
+
+    tenant: str = Field(
+        None,
+        title='Name of the Workday tenant',
+        description='Name of the Workday tenant',
+        example='my_tenant'
+    )
+    username: str = Field(
+        None,
+        title='Username for authentification',
+        description='Username for Credential authentification. Leave empty for Anonymous authentification.',
+        example='my_username'
+    )
+    password: str = Field(
+        None,
+        title='Password for authentification',
+        description='Password for Credential authentification',
+        example='$ecreTP4s$w0rd!'
+    )
+
+
+
+
+    def _retrieve_data(self, data_source: WorkdayDataSource) -> pd.DataFrame:
+		""" Only supports Anonymous or Credentials authentification """
+
+		if self.username:
+			auth = WsSecurityCredentialAuthentication('@'.join((self.username, self.tenant)), self.password)
+		else:
+			auth = AnonymousAuthentication()
+
+		apis = {
+			data_source.service: data_source.service_WSDL_URL
+		}
+
+		client = workday.WorkdayClient(
+			wsdls=apis,
+			authentication=auth
+		)
+		#print(client)
+
+
+		query_arguments = dict()
+		if not (data_source.request_references_param is None) and len(data_source.request_references_param) > 0:
+			query_arguments.update(Request_References = data_source.request_references_param)
+		if not (data_source.request_criteria_param is None) and len(data_source.request_criteria_param) > 0:
+			query_arguments.update(Request_Criteria = data_source.request_criteria_param)
+		if not (data_source.response_filter_param is None) and len(data_source.response_filter_param) > 0:
+			query_arguments.update(Response_Filter = data_source.response_filter_param)
+		if not (data_source.response_group_param is None) and len(data_source.response_group_param) > 0:
+			query_arguments.update(Response_Group = data_source.response_group_param)
```

Comment:
I don't understand this code, could you tell me why you are doing this ? is there not a simpler way to do it ? if not could you comment ?

Reply:
We deleted this part